ObservatoryFramework/EventArgs.cs

```diff
@@ -69,4 +69,39 @@ public enum NotificationRendering
         PluginNotifier = 4,
         All = (NativeVisual | NativeVocal | PluginNotifier)
     }
+
+    public enum LogMonitorState
+    {
+        None, // This is used only for startup.
+        Idle, // Monitoring is stopped
+        ReadAllBatch, // Performing a batch read of history. Formerly known as "read-all".
+        PreReadBatch, // Currently pre-reading current system context, if enabled, prior to RealTime monitoring
+        Realtime, // Real-time monitoring is active
+    }
```

Comment:
I also considered this:

```
    [Flags]
    public enum LogMonitorState : uint
    {
        Idle = 0, // Monitoring is stopped
        Realtime = 1, // Real-time monitoring is active
        Batch = 2, // Performing a some form of batch read of history
        PreRead = 4 + 2, // Currently pre-reading current system context (a Batch read)
        ReadAll = 8 + 2, // Currently re-reading all history (a Batch read).
    }
```

This would simplify the check for batch reads arguably, but add a value that's not otherwise explicitly used. So opted not to.

That said, using a flags enum, when checking state, one can no longer just check equality (ie. `currentState == LogMonitorState.Realtime`) -- one must use a mask. An equality would check will fail while actually monitoring if checked while also reading-all. If we're ok with that, let's proceed.